### ISSUE TITLE
malformed LinuxFxVersion on azurerm_linux_function_app with docker

### ISSUE THREAD

**Issue by peanutsguy** [hashicorp/terraform-provider-azurerm#19955]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.4

### AzureRM Provider Version

3.38

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app

### Terraform Configuration Files

```hcl
resource "azurerm_app_service_plan" "this" {
  name                = "funasp-${var.azurerm_function_app_name}-${var.enviroment}-${var.abrloc}-${var.resno}"
  location            = var.location
  resource_group_name = var.rgname
  kind = "elastic"
  sku {
    tier = "ElasticPremium"
    size = "EP1"
  }
}

resource "azurerm_linux_function_app" "this" {
  name                = "fun-${var.azurerm_function_app_name}-${var.enviroment}-${var.abrloc}-${var.resno}"
  location                   = var.location
  resource_group_name        = var.rgname
  service_plan_id        = azurerm_app_service_plan.this.id
  storage_account_name       = var.storage_name
  storage_account_access_key = var.storage_key
  functions_extension_version = "~4"
  site_config {
    container_registry_use_managed_identity = false
    application_stack {
      docker {
        registry_url = "{someregistryurl}.azurecr.io"
        image_name = "{someimage}"
        image_tag = "{sometag}"
        registry_username = var.registry_username
        registry_password = var.registry_password
      }
    }
  }
  app_settings =  {
    APPINSIGHTS_INSTRUMENTATIONKEY = var.instrumentation_key
    WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
  }
  identity {
    type = "SystemAssigned"
  }
  tags = var.tags
}
```

### Debug Output/Panic Output

```shell
│ Error: creating Linux Function App: (Site Name "{somefunctionname}" / Resource Group "{someresourcegroup}"): web.AppsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="The parameter LinuxFxVersion has an invalid value." Details=[{"Message":"The parameter LinuxFxVersion has an invalid value."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","ExtendedCode":"01007","Message":"The parameter LinuxFxVersion has an invalid value.","MessageTemplate":"The parameter {0} has an invalid value.","Parameters":["LinuxFxVersion"]}}]
│ 
│   with module.functionapp01.azurerm_linux_function_app.this,
│   on ../modules/DockerFunctionApp/main.tf line 38, in resource "azurerm_linux_function_app" "this":
│   38: resource "azurerm_linux_function_app" "this" {
│ 
│ creating Linux Function App: (Site Name "{somefunctionname}" /
│ Resource Group "{someresourcegroup}"):
│ web.AppsClient#CreateOrUpdate: Failure sending request: StatusCode=400 --
│ Original Error: Code="BadRequest" Message="The parameter LinuxFxVersion has
│ an invalid value." Details=[{"Message":"The parameter LinuxFxVersion has an
│ invalid
│ value."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","ExtendedCode":"01007","Message":"The
│ parameter LinuxFxVersion has an invalid value.","MessageTemplate":"The
│ parameter {0} has an invalid value.","Parameters":["LinuxFxVersion"]}}]
```

### Expected Behaviour

An Azure Function with a Docker should be created

### Actual Behaviour

The deployment fails with this error message:
`The parameter LinuxFxVersion has an invalid value.`

### Steps to Reproduce

Basically run `terraform apply`

I've tried setting `registry_url` with the following formats:
- `https://{someregistryurl}.azurecr.io`
- `https://{someregistryurl}.azurecr.io/{someimage}`
- `https://{someregistryurl}.azurecr.io/{someimage}:{sometag}`
- `{someregistryurl}.azurecr.io`
- `{someregistryurl}.azurecr.io/{someimage}`
- `{someregistryurl}.azurecr.io/{someimage}:{sometag}`

Also, I've tried setting `registry_username` and `registry_password` as `app_settings`

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/16290
https://github.com/hashicorp/terraform-provider-azurerm/issues/16483

**Comment by xiaxyi** [comment]
Thanks @peanutsguy for raising this issue, we are currently checking it and you may refer to pr:https://github.com/hashicorp/terraform-provider-azurerm/pull/18194 for the fix progress.

**Comment by xiaxyi** [comment]
@peanutsguy can you let me know the malformed value of linuxFxVersion? have you tried the latest provider?

**Comment by peanutsguy** [comment]
Let me try it and I'll get back to you

**Comment by rcskosir** [comment]
@peanutsguy Thank you for taking the time to open this issue. I am going to mark this issue as resolved with the release of #18194. If you believe this to be an error, please reach out and we can reopen. Thanks!
